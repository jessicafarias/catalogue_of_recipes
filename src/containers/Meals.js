import { useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Slider from '../components/Slider';
import { fetchMealsAction } from '../actions';
import getRecipes from '../requests/getRecipes';

const Meals = ({ meals, fetch }) => {
  const { category } = useParams();

  useEffect(() => {
    getRecipes(category).then(res => {
      fetch(res.meals);
    });
  }, []);

  return (
    <div>
      <Slider input={meals} />
    </div>
  );
};

Meals.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.shape({
    idMeal: PropTypes.string,
    strMealThumb: PropTypes.string,
    strMeal: PropTypes.string,
  })).isRequired,
  fetch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  meals: state.meals,
});

const mapDispatchToProps = dispatch => ({
  fetch: meals => {
    dispatch(fetchMealsAction(meals));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Meals);
