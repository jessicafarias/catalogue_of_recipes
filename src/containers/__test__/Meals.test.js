import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import rootReducer from '../../reducers';
import Meals from '../Meals';

const store = createStore(rootReducer, applyMiddleware(thunk));

describe('Display Meals.js', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Meals />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
