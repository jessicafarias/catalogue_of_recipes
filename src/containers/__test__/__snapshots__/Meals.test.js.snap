// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Meals.js renders correctly 1`] = `
<div>
  <div
    className="m-0 p-0"
  >
    <div
      className=""
    >
      <div
        className="carousel slide m-0"
        data-bs-interval="false"
        data-interval="false"
        data-ride="carousel"
        id="slide_in1"
      >
        <ol
          className="carousel-indicators"
        >
          <li
            className="active"
            data-slide-to={0}
            data-target="#slide_in0"
          />
        </ol>
        <div
          className="carousel-inner"
        >
          <div
            className="carousel-item active s1955165876"
          >
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".s1955165876  #uberuns:after {
 content:''; background: url(\\"https://www.themealdb.com/images/media/meals/1529444830.jpg\\"); background-attachment:fixed; background-size:cover; }
",
                }
              }
              type="text/css"
            />
            <div
              className="image"
              style={
                Object {
                  "backgroundImage": "url(https://www.themealdb.com/images/media/meals/1529444830.jpg)",
                }
              }
            />
            <div
              className="section"
              data="url(https://www.themealdb.com/images/media/meals/1529444830.jpg)"
              id="uberuns"
            >
              <p
                className="text_slider"
              >
                Title
              </p>
              <p
                className="span_title pb-2"
              >
                By Themealdb
              </p>
              <div
                className="row m-2 p-2"
              >
                <div
                  className="col-4 d-flex justify-content-center align-items-center"
                >
                  <svg
                    className="bi bi-star-fill"
                    fill="#e9d061"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                    />
                  </svg>
                  <p
                    className="tag_slider"
                  >
                    FAVORITE
                  </p>
                </div>
                <div
                  className="col-4 d-flex justify-content-center align-items-center"
                >
                  <svg
                    className="bi bi-heart-fill"
                    fill="#e9d061"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 1.314C12.438-3.248 23.534 4.735 8 15-7.534 4.736 3.562-3.248 8 1.314z"
                      fillRule="evenodd"
                    />
                  </svg>
                  <p
                    className="tag_slider"
                  >
                    RECOMMEND
                  </p>
                </div>
                <div
                  className="col-4 d-flex justify-content-center align-items-center"
                >
                  <svg
                    className="bi bi-share-fill"
                    fill="#e9d061"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11 2.5a2.5 2.5 0 1 1 .603 1.628l-6.718 3.12a2.499 2.499 0 0 1 0 1.504l6.718 3.12a2.5 2.5 0 1 1-.488.876l-6.718-3.12a2.5 2.5 0 1 1 0-3.256l6.718-3.12A2.5 2.5 0 0 1 11 2.5z"
                    />
                  </svg>
                  <p
                    className="tag_slider"
                  >
                    SHARE
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          data-slide="prev"
          href="#slide_in1"
          role="button"
        >
          <span
            aria-hidden="false"
            className="carousel-control-prev-icon"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          data-slide="next"
          href="#slide_in1"
          role="button"
        >
          <span
            aria-hidden="false"
            className="carousel-control-next-icon"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
